File: c/grade-check.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char * argv[]) {
    FILE* fp;
    if(argc < 3) {
        printf("Provide a student and grades file\n");
    }
    else {
        char * student = argv[1];
        char * file = argv[2];

        fp = fopen(file, "r");
        if(fp){
            char grades[1095];
            fgets(grades, 1095, (FILE *)fp);
            char * res = strstr(grades, student);
            if(res != NULL)
                printf("%s", res);
            else
                printf("No grade found for %s", student);
        }
        else{
            printf("Unable to open file\n");
        }
    }
    return 0;
}

File: src/bin/grade-check.rs
use std::{fs,env};

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() < 3 {
        println!("Provide a student name and grades file");
    }
    else{
        let student = &args[1].to_lowercase();
        let file = &args[2]; 

        println!("Opening file: {}", file);
        let grades = fs::read_to_string(file);

        match grades {
            Ok(_) => {
                let student_grade = grades.as_ref().unwrap().find(student);
                if let Some(res) = student_grade {
                    let grade = &grades.unwrap()[res..(res+&student.len()+2)];
                    println!("{}", grade);
                }
                else {
                    println!("No grade for {}", student);
                }
            },
            Err(_) => println!("Unable to open file")
        }

    }

}